Wizardry/Core/BattleSys/Source/BattleHit.c

```diff
@@ -569,8 +569,34 @@ STATIC_DECLAR int BattleHit_CalcDamage(struct BattleUnit * attacker, struct Batt
         decrease += DAMAGE_DECREASE(SKILL_EFF0(SID_CrusaderWard));
 #endif
 
+    bool barricadePlus_activated = false;
+
+#if (defined(SID_BarricadePlus) && (COMMON_SKILL_VALID(SID_BarricadePlus)))
+    if (BattleSkillTester(defender, SID_BarricadePlus))
+    {
+        if (defender->unit.curHP < defender->hpInitial)
```

Comment:
seems no need on such judgement

Reply:
True, since we're already checking for the number of attacks the skill holder has sustained.